app/controllers/file.rb

```diff
@@ -57,49 +57,40 @@ class File < Base
 						if new.exist?
 							return slim :error, locals: { message: "A file with that name already exists" }
 						else
-							begin
-								site.git.lib.mv(file.relative_path.to_s, new.relative_path.to_s)
-								commit(site.git, "Renamed #{file.relative_path} to #{new.name}")
-							rescue Git::GitExecuteError
-								# Not in git, so just rename
-								file.rename(new.path)
+							site.commit_with("Renamed #{file.relative_path} to #{new.name}", pundit_user) do |dir|
+								dir.join(file.relative_path).rename(dir.join(new.relative_path))
 							end
 						end
 					elsif params[:source]
-						file.write params[:source].encode(universal_newline: true)
-
-						add_and_commit(site, file, "Modified #{file.relative_path}")
+						site.commit_with("Modified #{file.relative_path}", pundit_user) do |dir|
+							dir.join(file.relative_path).write params[:source].encode(universal_newline: true)
```

Comment:
Oh I see, `dir` is not the same directory, we are making a new temporary directory (if I'm reading correctly)